### PR TITLE
Moved ID export line for logging reasons

### PR DESCRIPTION
**Pull request by ghaabor**
Wanted to log HMAC ids for unsuccessful responses too, so I've moved this line before returning anything.

### CODE REVIEW

**Comment by ghaabor** [comment]
@pwolanin ^

**Comment by ghaabor** [comment]
pingin @pwolanin again

File: lib/acquia-http-hmac/rack_authenticate.rb [hunk 1]
```diff
@@ -23,15 +23,20 @@ def call(env)
         return unauthorized if auth_header.empty?
 
         attributes = Acquia::HTTPHmac::Auth.parse_auth_header(auth_header)
+
+        # Pass the id to later stages
+        # Replaced id here to be able to log invalid requests too
+        env['ACQUIA-HTTP-HMAC-ID'] = attributes[:id]
```
Comment (pdrakeweb) [inline lib/acquia-http-hmac/rack_authenticate.rb:29]: I believe that ACQUIA-HTTP-HMAC-ID is intended to indicate that the current execution has authenticated as that ID.  Moving it here will break code that makes this assumption.  A different environment variable could be used to store the attempted authentication ID prior to validating the authentication attempt.

Comment (ghaabor) [reply lib/acquia-http-hmac/rack_authenticate.rb:29]: @pdrakeweb That's a good idea, I'll send in a modification later today.

Comment (ghaabor) [reply lib/acquia-http-hmac/rack_authenticate.rb:29]: @pdrakeweb closing this PR in favor of this: https://github.com/acquia/http-hmac-ruby/pull/15

**Pull request closed without merging**